projects/packages/forms/src/service/class-hostinger-reach-integration.php

```diff
@@ -0,0 +1,255 @@
+			} elseif ( ( $id === 'lastname' || $label === 'lastname' ) && ! empty( $value ) ) {
+				$subscriber_data['surname'] = $value;
+			}
+		}
+
+		if ( empty( $subscriber_data['email'] ) ) {
+			return array();
+		}
+
+		return $subscriber_data;
+	}
+
+	/**
+	 * Check if submission has consent when required.
+	 *
+	 * @param Feedback $feedback Feedback object for the submission.
+	 * @param mixed    $form     Contact form instance.
+	 * @param bool     $is_v2    Whether the data comes from v2 storage.
+	 * @return bool True if consent is present or not required; false otherwise.
+	 */
+	protected static function has_consent( $feedback, $form, $is_v2 ) {
+		if ( $is_v2 ) {
+			if ( $feedback->has_field_type( 'consent' ) && ! $feedback->has_consent() ) {
+				return false;
+			}
+			return true;
+		}
+
+		$consent_field = null;
+		if ( is_object( $form ) && is_array( $form->fields ) ) {
+			foreach ( $form->fields as $form_field ) {
+				if ( 'consent' === $form_field->get_attribute( 'type' ) ) {
+					$consent_field = $form_field;
+					break;
+				}
+			}
+		}
+		if ( $consent_field ) {
+			$consent_type = strtolower( (string) $consent_field->get_attribute( 'consenttype' ) );
+			if ( 'explicit' === $consent_type && ! $consent_field->value ) {
+				return false;
+			}
+		}
+
+		return true;
+	}
+
+	/**
+	 * Handle Hostinger Reach integration after feedback post is inserted.
+	 *
+	 * For now, this only validates preconditions and returns early; real actions will be implemented later.
+	 *
+	 * @param int   $post_id The post ID for the feedback CPT.
+	 * @param array $fields  Collection of Contact_Form_Field instances.
+	 * @param bool  $is_spam Whether the submission is spam.
+	 */
+	public static function handle_hostinger_reach_integration( $post_id, $fields, $is_spam ) {
+		if ( $is_spam ) {
+			return;
+		}
+
+		// Try and get the form from any of the fields
+		$form = null;
+		foreach ( $fields as $field ) {
+			if ( ! empty( $field->form ) ) {
+				$form = $field->form;
+				break;
+			}
+		}
+		if ( ! $form || ! is_a( $form, 'Automattic\\Jetpack\\Forms\\ContactForm\\Contact_Form' ) ) {
+			return;
+		}
+
+		// Feature is toggled per-form.
+		if ( empty( $form->attributes['hostingerReach']['enabledForForm'] ?? null ) ) {
+			return;
+		}
+
+		$feedback = Feedback::get( $post_id );
+		if ( ! $feedback ) {
+			return;
+		}
+
+		$api_handler = self::get_api();
+		if ( ! $api_handler ) {
+			return;
+		}
+
+		// Respect consent if a consent field exists.
+		$post       = get_post( $post_id );
+		$is_v2_data = ( $post && $post->post_mime_type === 'v2' );
+		if ( ! self::has_consent( $feedback, $form, $is_v2_data ) ) {
+			return;
+		}
+
+		$subscriber_data = $is_v2_data ? self::get_subscriber_data( $feedback ) : self::get_subscriber_data_from_fields( $fields );
+		if ( empty( $subscriber_data ) ) {
+			return;
+		}
+
+		$group_name = self::get_group_name( $form );
+
+		self::submit_contact( $api_handler, $subscriber_data, $group_name );
+	}
+}
```

Comment:
This is the class that handles the actual integration with Hostinger Reach when a form is submitted. There's a main `handle_hostinger_reach_integration` method that does some checks and then calls the other methods as helpers to instantiate the API, get subscriber data and group name, and submit the form.